### DISCUSSION TITLE
How do I define input for tools?

### DISCUSSION THREAD

**Discussion by Harrolee** [microsoft/autogen#4434] [Q&A] [answered]
Hey! Apologies if this is explained somewhere. I haven't been able to find it in the v0.4.0 tutorials or examples.

How would I tell an AssistantAgent which input to provide to a tool? I want my AssistantAgent to extract information from conversation history and use that information as input to a tool. How do I define the sequence of tool-calling events? Should I create a new class that inherits from BaseChatAgent?

My mental model of the AssistantAgent flow is this:
1. AssistantAgent makes a call to an LLM with a prompt that contains its description, name, tools, and conversation history
2. The LLM responds with a description of which tool to use and how to use it
3. AssistantAgent submits the LLM's response to another LLM, which actuates the operation

Is this accurate? Please tell me how it actually works if not.

Thanks,
Lee

**Comment by Harrolee** [comment]
I'll pop into office hours on Wednesday

**Comment by ekzhu** [comment, marked as answer]
The assistant agent uses the model to generate input to the tools. Caller cannot predetermine the sequence of tool calls -- if that's the goal it's better to use a team such as round robin group chat to specify list of agents that will be called in sequence.

**Reply by Harrolee** [reply]
Gotcha, thanks ekzhu!

**Reply by Harrolee** [reply]
"The assistant agent uses the model to generate input to the tools."
The tool description and assistant agent description are the criteria that the model uses to create input for the tools. 
[Here](https://github.com/microsoft/autogen/blob/f02aac78fb47818c378407cd4578fc7b82703597/python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py#L289), right?

**Reply by Harrolee** [reply]
(just confirming that I understand what you said)

**Reply by ekzhu** [reply]
The assistant system message and the tool description and function signature is used to determine the tool calls and their inputs. 

The description field is used by team's orchestration logic to select agent.

**Reply by Harrolee** [reply]
Thank you for that distinction!
